### ISSUE TITLE
You forgot to add 'mini-css-extract-plugin' plugin

### ISSUE THREAD

**Issue by himself65** [vanilla-extract-css/vanilla-extract#1101]
### Describe the bug

in `@vanilla-extract/next-plugin` 2.1.3

### Reproduction

https://github.com/toeverything/AFFiNE

```
yarn install
cd apps/web
yarn up @vanilla-extract/next-plugin@2.1.3
yarn dev
```

### System Info

```shell
➜  AFFiNE git:(master) ✗ npx envinfo --system --npmPackages '{@vanilla-extract/*,webpack,esbuild,vite,rollup}' --binaries --browsers

  System:
    OS: macOS 13.4
    CPU: (8) arm64 Apple M2
    Memory: 1.15 GB / 24.00 GB
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 18.16.0 - ~/.nvm/versions/node/v18.16.0/bin/node
    Yarn: 3.5.0 - ~/.nvm/versions/node/v18.16.0/bin/yarn
    npm: 9.5.1 - ~/.nvm/versions/node/v18.16.0/bin/npm
  Browsers:
    Chrome: 113.0.5672.126
    Firefox: 110.0.1
    Safari: 16.5
  npmPackages:
    @vanilla-extract/vite-plugin: ^3.8.2 => 3.8.2 
    vite: ^4.3.9 => 4.3.9 
```

### Used Package Manager

yarn

### Logs

```shell
│  AFFiNE git:(master) ✗ yarn dev
◆  Select dev server
◇  Select dev server
│  local - 127.0.0.1:3000
│
◇  Debug blocksuite locally?
│  No
You might need setup OctoBase dev server first.
- ready started server on 0.0.0.0:8080, url: http://localhost:8080
- info Loaded env from /Users/himself65/Code/AFFiNE/apps/web/.env.local
Build Flags {
  enableAllPageFilter: false,
  enableImagePreviewModal: true,
  enableTestProperties: true,
  enableLegacyCloud: false,
  enableBroadCastChannelProvider: true,
  enableDebugPage: true,
  changelogUrl: 'https://affine.pro/blog/whats-new-affine-0518'
}
Editor Flags {
  enable_database: true,
  enable_slash_menu: true,
  enable_edgeless_toolbar: true,
  enable_block_hub: true,
  enable_drag_handle: true,
  enable_surface: true,
  enable_linked_page: true,
  enable_bookmark_operation: true
}
NEXT_PUBLIC_FIREBASE_API_KEY found.
SENTRY_AUTH_TOKEN not found, Sentry monitoring feature will be disabled.
perfsee token not found. performance monitoring will be disabled.
- warn You have enabled experimental feature (swcPlugins) in next.config.mjs.
- warn Experimental features are not covered by semver, and may cause unexpected or broken application behavior. Use at your own risk.

API request proxy to [local Server]: 127.0.0.1:3000
- error ../../packages/component/src/packages/component/src/components/affine-banner/index.css.ts.vanilla.css!=!../../node_modules/@vanilla-extract/webpack-plugin/virtualFileLoader/dist/vanilla-extract-webpack-plugin-virtualFileLoader.cjs.js?{"fileName":"../../packages/component/src/components/affine-banner/index.css.ts.vanilla.css","source":"#H4sIAAAAAAAAE7VUO2/bMBDe8yu4FGgHBn4lcZWlaLp06NQCHQ1KPMlX06RAUrKVIv+9JypKLEux1RodZME88r4Hv9OnDVSpFVtwTKQpauCx0BrsyimU8MXs9Gpl7rJlOZuw31eMTd6FF2NrwGztIzbJ9/e08ETPdNKrLhZt+enqugsQW7NzYH8Kq1Fn332loIWahi6xSDaZNYWWPDHK2IiVwr7n/KXNS3nX9Gi2fajxhg70d7Us57cNy9Rozx0+wtHJZn0bzkh0uRJVxFIF4dCvwnlMK2qrPWjqltAv2LokFGaao4etO1zOjUOPRkfMghIeS7jv25Mo4+Bz4b3RHW9mwZsdSr9+JX4sZEg+EsFDhwrr6j25wZbXeWl1gYOWo8SJ2BlVeKhXbcNvGvgNa/2aHCmdHyqd3nSVtv+HvGTskaOWsKdtA2iSQq2MkD8wfyBhgkq2A7wYE7/c4lbY6lzm1mRQqLY6aEYOdbQDcjJ6sXDwP8NHmzWpeb62c5+B67ljQIzIY24KT9QtjZZ0p63uOHwTHL5YzSm8XpxuA+aoS5otumGb9S+pXbogfw9vTPjdCKL/NL1/kY2B2W0Uv6nmGzgnMugoWY5QQkOUUZAUpISxHIRQqDedvh9H9PWw91xCYuxzrmmQwVKroY9tHyFamxJswLm0U4nkKshLOf8BqnyASKoHAAA="}!../../node_modules/@vanilla-extract/webpack-plugin/extracted.js
Error: You forgot to add 'mini-css-extract-plugin' plugin (i.e. `{ plugins: [new MiniCssExtractPlugin()] }`), please read https://github.com/webpack-contrib/mini-css-extract-plugin#getting-started
Import trace for requested module:
../../packages/component/src/packages/component/src/components/affine-banner/index.css.ts.vanilla.css!=!../../node_modules/@vanilla-extract/webpack-plugin/virtualFileLoader/dist/vanilla-extract-webpack-plugin-virtualFileLoader.cjs.js?{"fileName":"../../packages/component/src/components/affine-banner/index.css.ts.vanilla.css","source":"#H4sIAAAAAAAAE7VUO2/bMBDe8yu4FGgHBn4lcZWlaLp06NQCHQ1KPMlX06RAUrKVIv+9JypKLEux1RodZME88r4Hv9OnDVSpFVtwTKQpauCx0BrsyimU8MXs9Gpl7rJlOZuw31eMTd6FF2NrwGztIzbJ9/e08ETPdNKrLhZt+enqugsQW7NzYH8Kq1Fn332loIWahi6xSDaZNYWWPDHK2IiVwr7n/KXNS3nX9Gi2fajxhg70d7Us57cNy9Rozx0+wtHJZn0bzkh0uRJVxFIF4dCvwnlMK2qrPWjqltAv2LokFGaao4etO1zOjUOPRkfMghIeS7jv25Mo4+Bz4b3RHW9mwZsdSr9+JX4sZEg+EsFDhwrr6j25wZbXeWl1gYOWo8SJ2BlVeKhXbcNvGvgNa/2aHCmdHyqd3nSVtv+HvGTskaOWsKdtA2iSQq2MkD8wfyBhgkq2A7wYE7/c4lbY6lzm1mRQqLY6aEYOdbQDcjJ6sXDwP8NHmzWpeb62c5+B67ljQIzIY24KT9QtjZZ0p63uOHwTHL5YzSm8XpxuA+aoS5otumGb9S+pXbogfw9vTPjdCKL/NL1/kY2B2W0Uv6nmGzgnMugoWY5QQkOUUZAUpISxHIRQqDedvh9H9PWw91xCYuxzrmmQwVKroY9tHyFamxJswLm0U4nkKshLOf8BqnyASKoHAAA="}!../../node_modules/@vanilla-extract/webpack-plugin/extracted.js
../../packages/component/src/components/affine-banner/index.css.ts
../../packages/component/src/components/affine-banner/download-client.tsx
../../packages/component/src/components/affine-banner/index.tsx
./src/components/blocksuite/workspace-header/header.tsx
./src/components/blocksuite/workspace-header/index.tsx
./src/components/page-detail-editor.tsx
./src/adapters/local/index.tsx
./src/adapters/workspace.ts
./src/hooks/affine/use-affine-log-out.ts
./src/components/pure/message-center/index.tsx
- wait compiling...
API request proxy to [local Server]: 127.0.0.1:3000
- wait compiling / (client and server)...
Build Flags {
  enableAllPageFilter: false,
  enableImagePreviewModal: true,
  enableTestProperties: true,
  enableLegacyCloud: false,
  enableBroadCastChannelProvider: true,
  enableDebugPage: true,
  changelogUrl: 'https://affine.pro/blog/whats-new-affine-0518'
}
Editor Flags {
  enable_database: true,
  enable_slash_menu: true,
  enable_edgeless_toolbar: true,
  enable_block_hub: true,
  enable_drag_handle: true,
  enable_surface: true,
  enable_linked_page: true,
  enable_bookmark_operation: true
}
NEXT_PUBLIC_FIREBASE_API_KEY found.
SENTRY_AUTH_TOKEN not found, Sentry monitoring feature will be disabled.
perfsee token not found. performance monitoring will be disabled.
- error ../../packages/component/src/packages/component/src/components/affine-banner/index.css.ts.vanilla.css!=!../../node_modules/@vanilla-extract/webpack-plugin/virtualFileLoader/dist/vanilla-extract-webpack-plugin-virtualFileLoader.cjs.js?{"fileName":"../../packages/component/src/components/affine-banner/index.css.ts.vanilla.css","source":"#H4sIAAAAAAAAE7VUO2/bMBDe8yu4FGgHBn4lcZWlaLp06NQCHQ1KPMlX06RAUrKVIv+9JypKLEux1RodZME88r4Hv9OnDVSpFVtwTKQpauCx0BrsyimU8MXs9Gpl7rJlOZuw31eMTd6FF2NrwGztIzbJ9/e08ETPdNKrLhZt+enqugsQW7NzYH8Kq1Fn332loIWahi6xSDaZNYWWPDHK2IiVwr7n/KXNS3nX9Gi2fajxhg70d7Us57cNy9Rozx0+wtHJZn0bzkh0uRJVxFIF4dCvwnlMK2qrPWjqltAv2LokFGaao4etO1zOjUOPRkfMghIeS7jv25Mo4+Bz4b3RHW9mwZsdSr9+JX4sZEg+EsFDhwrr6j25wZbXeWl1gYOWo8SJ2BlVeKhXbcNvGvgNa/2aHCmdHyqd3nSVtv+HvGTskaOWsKdtA2iSQq2MkD8wfyBhgkq2A7wYE7/c4lbY6lzm1mRQqLY6aEYOdbQDcjJ6sXDwP8NHmzWpeb62c5+B67ljQIzIY24KT9QtjZZ0p63uOHwTHL5YzSm8XpxuA+aoS5otumGb9S+pXbogfw9vTPjdCKL/NL1/kY2B2W0Uv6nmGzgnMugoWY5QQkOUUZAUpISxHIRQqDedvh9H9PWw91xCYuxzrmmQwVKroY9tHyFamxJswLm0U4nkKshLOf8BqnyASKoHAAA="}!../../node_modules/@vanilla-extract/webpack-plugin/extracted.js
Error: You forgot to add 'mini-css-extract-plugin' plugin (i.e. `{ plugins: [new MiniCssExtractPlugin()] }`), please read https://github.com/webpack-contrib/mini-css-extract-plugin#getting-started
Import trace for requested module:
../../packages/component/src/packages/component/src/components/affine-banner/index.css.ts.vanilla.css!=!../../node_modules/@vanilla-extract/webpack-plugin/virtualFileLoader/dist/vanilla-extract-webpack-plugin-virtualFileLoader.cjs.js?{"fileName":"../../packages/component/src/components/affine-banner/index.css.ts.vanilla.css","source":"#H4sIAAAAAAAAE7VUO2/bMBDe8yu4FGgHBn4lcZWlaLp06NQCHQ1KPMlX06RAUrKVIv+9JypKLEux1RodZME88r4Hv9OnDVSpFVtwTKQpauCx0BrsyimU8MXs9Gpl7rJlOZuw31eMTd6FF2NrwGztIzbJ9/e08ETPdNKrLhZt+enqugsQW7NzYH8Kq1Fn332loIWahi6xSDaZNYWWPDHK2IiVwr7n/KXNS3nX9Gi2fajxhg70d7Us57cNy9Rozx0+wtHJZn0bzkh0uRJVxFIF4dCvwnlMK2qrPWjqltAv2LokFGaao4etO1zOjUOPRkfMghIeS7jv25Mo4+Bz4b3RHW9mwZsdSr9+JX4sZEg+EsFDhwrr6j25wZbXeWl1gYOWo8SJ2BlVeKhXbcNvGvgNa/2aHCmdHyqd3nSVtv+HvGTskaOWsKdtA2iSQq2MkD8wfyBhgkq2A7wYE7/c4lbY6lzm1mRQqLY6aEYOdbQDcjJ6sXDwP8NHmzWpeb62c5+B67ljQIzIY24KT9QtjZZ0p63uOHwTHL5YzSm8XpxuA+aoS5otumGb9S+pXbogfw9vTPjdCKL/NL1/kY2B2W0Uv6nmGzgnMugoWY5QQkOUUZAUpISxHIRQqDedvh9H9PWw91xCYuxzrmmQwVKroY9tHyFamxJswLm0U4nkKshLOf8BqnyASKoHAAA="}!../../node_modules/@vanilla-extract/webpack-plugin/extracted.js
../../packages/component/src/components/affine-banner/index.css.ts
../../packages/component/src/components/affine-banner/browser-warning.tsx
../../packages/component/src/components/affine-banner/index.tsx
./src/components/blocksuite/workspace-header/header.tsx
./src/components/blocksuite/workspace-header/index.tsx
./src/components/page-detail-editor.tsx
./src/adapters/affine/index.tsx
./src/adapters/workspace.ts
./src/hooks/affine/use-affine-log-out.ts
./src/components/pure/message-center/index.tsx
Build Flags {
  enableAllPageFilter: false,
  enableImagePreviewModal: true,
  enableTestProperties: true,
  enableLegacyCloud: false,
  enableBroadCastChannelProvider: true,
  enableDebugPage: true,
  changelogUrl: 'https://affine.pro/blog/whats-new-affine-0518'
}
Editor Flags {
  enable_database: true,
  enable_slash_menu: true,
  enable_edgeless_toolbar: true,
  enable_block_hub: true,
  enable_drag_handle: true,
  enable_surface: true,
  enable_linked_page: true,
  enable_bookmark_operation: true
}
NEXT_PUBLIC_FIREBASE_API_KEY found.
SENTRY_AUTH_TOKEN not found, Sentry monitoring feature will be disabled.
perfsee token not found. performance monitoring will be disabled.
API request proxy to [local Server]: 127.0.0.1:3000
API request proxy to [local Server]: 127.0.0.1:3000
- wait compiling /_error (client and server)...
```

### Validations

- [X] Check that there isn't [already an issue](https://github.com/vanilla-extract-css/vanilla-extract/issues) that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by Himself65** [comment]
https://github.com/toeverything/AFFiNE/commit/e0eb216b9b88d323b91731a5c2fc735164a7c340

**Comment by askoufis** [comment]
@Himself65 I wouldn't exactly classify that repo as a reproduction. Ideally it should be a small as possible and should clearly isolate the issue to a specific change. That being said, it looks you weren't enabling the `appDir` before updating to `2.1.3`, but now that it's using the resolved next config, `appDir` will default to `true`, so that's potentially the cause of the issue.

**Comment by graup** [comment]
@shuding looks like there's still some weirdness going on with the appDir config, mind taking another look here?

**Comment by Himself65** [comment]
> I wouldn't exactly classify that repo as a reproduction. Ideally it should be as small as possible and should clearly isolate the issue to a specific change.

Sorry for I don't have much time to create a mini reproduction repo

**Comment by Sharlottes** [comment]
when i upgrade to `@vanilla-extract/next-plugin@v2.2.0` from `@vanilla-extract/next-plugin@v2.1.2`, `next dev` throws below error log and process died.

```bash
user@DESKTOP-18FQSK1 MINGW64 ~/Documents/GitHub/Homepage (dev)
$ yarn dev
yarn run v1.22.19
$ next dev
- ready started server on 0.0.0.0:3000, url: http://localhost:3000
TypeError: Right-hand side of 'instanceof' is not an object
    at C:\Users\user\Documents\GitHub\Homepage\node_modules\@vanilla-extract\next-plugin\dist\vanilla-extract-next-plugin.cjs.dev.js:109:47
    at Array.some (<anonymous>)
    at Object.webpack (C:\Users\user\Documents\GitHub\Homepage\node_modules\@vanilla-extract\next-plugin\dist\vanilla-extract-next-plugin.cjs.dev.js:109:25)
    at getBaseWebpackConfig (C:\Users\user\Documents\GitHub\Homepage\node_modules\next\dist\build\webpack-config.js:2147:32)
    at async Promise.all (index 2)
    at async Span.traceAsyncFn (C:\Users\user\Documents\GitHub\Homepage\node_modules\next\dist\trace\trace.js:103:20)
    at async Span.traceAsyncFn (C:\Users\user\Documents\GitHub\Homepage\node_modules\next\dist\trace\trace.js:103:20)
    at async HotReloader.start (C:\Users\user\Documents\GitHub\Homepage\node_modules\next\dist\server\dev\hot-reloader.js:573:30)
    at async DevServer.prepareImpl (C:\Users\user\Documents\GitHub\Homepage\node_modules\next\dist\server\dev\next-dev-server.js:685:9)
    at async NextServer.prepare (C:\Users\user\Documents\GitHub\Homepage\node_modules\next\dist\server\next.js:165:13)
    at async Server.<anonymous> (C:\Users\user\Documents\GitHub\Homepage\node_modules\next\dist\server\lib\render-server.js:136:17) {
  type: 'TypeError'
}
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

it's fixed when I downgrade to `@vanilla-extract/next-plugin@v2.1.3` again. because [the error line](https://github.com/vanilla-extract-css/vanilla-extract/pull/1105/files#diff-fe1f031aa9fe2619b9fa6c96c9879c91df7878306dded38a45ad67a55f067205R138) is newly added on #1105, which released on `@vanilla-extract/next-plugin@v2.2.0`.
![image](https://github.com/vanilla-extract-css/vanilla-extract/assets/60801210/338dc315-a670-4b47-b1d9-cb091cdc060e)

**Comment by aldoprogrammer** [comment]
I got same issue, and as it was saying to read the [guidline here](https://github.com/webpack-contrib/mini-css-extract-plugin#getting-started), then i follow it by installing using the command pnpm add -D mini-css-extract-plugin, then the problem is gone.
